pandas/tests/indexing/test_at.py

```diff
@@ -39,3 +39,19 @@ def test_at_with_duplicate_axes_requires_scalar_lookup(self):
             df.at[1, ["A"]] = 1
         with pytest.raises(ValueError, match=msg):
             df.at[:, "A"] = 1
+
+
+def test_at_assign_float_to_int_frame():
```

Comment:
can you parameterize this on .loc for the same examples (and move to pandas/test/test_loc.py)

Reply:
also testing iat/iloc for the same (obviously using the position). if its becomes messy then can do that in a followup.

Reply:
also we would want to test this for other dtypes as well. so that is for sure a followon, pls open an issue for this.

Reply:
all this said, we *might* have some tests for this already (likely we do), prob have to hunt a bit for them.

Reply:
Parametrized the tests for loc, iloc and iat would be an ugly parametrization. I will open an issue for dtypes and iloc/iat